internal/tfimport/importer/resources_test.go

```diff
@@ -75,3 +77,82 @@ func TestFromConfigValuesErr(t *testing.T) {
 		}
 	}
 }
+
+func TestUserValue(t *testing.T) {
+	tests := []struct {
+		input string
+		want  string
+	}{
+		// Empty.
+		{"\n", ""},
+
+		// Normal input.
+		{"myinput\n", "myinput"},
+
+		// Stripping spaces, but not in the middle.
+		{"  my space   separated  input   \n", "my space   separated  input"},
+	}
+
+	for _, tc := range tests {
+		// Temporarily redirect output to null while running
+		stdout := os.Stdout
+		os.Stdout = os.NewFile(0, os.DevNull)
+
+		out, err := userValue(strings.NewReader(tc.input))
+
+		// Restore.
+		os.Stdout = stdout
+
+		if err != nil {
+			t.Errorf("userValue(%q) failed: %s", tc.input, err)
+		} else if out != tc.want {
```

Comment:
You can just do `if out != tc.want` instead of else if, which is usually what people do in this case.

Reply:
I wanted to avoid printing both an error and a "output does not equal wanted". I'll change it and use `t.Fatalf()` above.